### DISCUSSION TITLE
Parsing and stringifying query strings

### DISCUSSION THREAD

**Discussion by MartinMalinda** [nuxt/nuxt#9659] [Questions (Nuxt 2)]
I have quite a few cases in my code where I have to parse and stringify query strings. There's libraries that do this well, but I'm wondering if I can import such functions from Nuxt itself because Nuxt is certainly doing this as part of its routing.

I know it's possible to import these functions from `vue-router`, but when I use Nuxt `vue-router` is encapsulating within it and I can't import from the `vue-router` package. Unless perhaps I would add `vue-router` as a dependency. But I'm a little bit worried about this. My intention is to save some extra code - don't include extra code that is already present in Nuxt. And this over time cause I would load vue-router twice.

Any tips? Thanks a lot!

**Comment by TheAlexLichter** [comment]
Hey Martin 👋

The idea of re-using code, especially `vue-router`'s stringify/resolve function sounds like a good one at first but consider these points:

1. You can't access these functionalities "easily". Because vue-router is written in flow, you can't simply import from source (e.g. `import { stringifyQuery } from 'vue-router/src/util/query.js'`). In the bundled version, the helper functions aren't exported anymore.
2. The code you need for that functionality shouldn't be too large. Don't worry about a few extra bytes (not kilobytes ofc) if you can test the code and know what it is doing. Ideally, you can even feed it into `vue-router` itself by overriding the default functions (see https://github.com/vuejs/vue-router/issues/1259#issuecomment-306000548)
3. If you need to resolve the current site's query, you can use `this.$route.query` already.
